controllers/imageupdateautomation_controller.go

```diff
@@ -78,22 +80,65 @@ type ImageUpdateAutomationReconciler struct {
 func (r *ImageUpdateAutomationReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	ctx := context.Background()
 	log := r.Log.WithValues("imageupdateautomation", req.NamespacedName)
+	now := time.Now()
 
 	var auto imagev1.ImageUpdateAutomation
 	if err := r.Get(ctx, req.NamespacedName, &auto); err != nil {
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
+	// whatever else happens, we've now "seen" the reconcile
+	// annotation if it's there
+	if token, ok := meta.ReconcileAnnotationValue(auto.GetAnnotations()); ok {
+		auto.Status.SetLastHandledReconcileRequest(token)
+		if err := r.Status().Update(ctx, &auto); err != nil {
+			return ctrl.Result{Requeue: true}, err
+		}
+	}
+
+	if auto.Spec.Suspend {
```

Comment:
We decided to not change the Ready status on suspend, instead just do https://github.com/fluxcd/kustomize-controller/blob/main/controllers/kustomization_controller.go#L167-L171

Reply:
Doesn't that break this: https://github.com/fluxcd/flux2/blob/main/cmd/flux/reconcile_kustomization.go#L107 ?

Reply:
If the idea is that it will time out and therefore (eventually) tell the user that the object wasn't reconciled, I would consider reporting that this is probably going happen up front when an object is suspended.

Reply:
Right, we need lookup the suspended status and tell the user that it should resume the reconciliation first.

Reply:
There's two choices for what happens when someone requests a reconciliation of a suspended resource:

 - time out (probably ... maybe someone else will unsuspend it while you're waiting)
 - trivially succeed

In both these cases it's worth telling the user up front that the resource is currently suspended. But elsewise, the difference is this: do you want the user to have to retry straight away; or is it worth waiting to see if something else changes the resource (they can always interrupt the program if they get bored waiting).

If reconciliation succeeds trivially, you at least know the machinery is running. Whereas if it times out, it could be because the controller is jammed somewhere. I don't think there's much utility in waiting to see if someone else unsuspends an object. For those reasons I prefer succeeding trivially, rather than timing out.

Reply:
Fixed in https://github.com/fluxcd/flux2/pull/525

Reply:
I should have had a third option, "bail out without trying". From the user's point of view, this is very similar to trivially succeeding, so :+1:

It doesn't make it any easier to test suspended resources, though -- for that, I recommend updating the reconcile request status before exiting Reconcile, so you can (programmatically) see that the loop has run.

Reply:
> I recommend updating the reconcile request status before exiting Reconcile

.. However in the absence of a convention, I've made my test run Reconcile explicitly.